src/task_processor.hpp

```diff
@@ -75,6 +82,16 @@ class TaskProcessor {
         threads_[i].join();
       }
     }
+    threads_rejoined_ = true;
+    // When all thread work completed and rejoined, check that no child threads had exceptions. 
```

Comment:
I'm going to suggest that this comment and the one in CheckExceptions could be eliminated if we renamed CheckExceptions to something like `RethrowAnyExceptionsEncounteredInChildThread`.

Or, perhaps this method could be eliminated? The code is very clear.

Reply:
I agree it is clear.  I just merged it into the end of Wait() and eliminated the method.